source/tyr/tyr/binarisation.py

```diff
@@ -127,10 +127,13 @@ def fusio2ed(self, instance_config, filename, job_id, dataset_uid):
 
     logger = get_instance_logger(instance)
     try:
-        working_directory = os.path.dirname(filename)
-
-        zip_file = zipfile.ZipFile(filename)
-        zip_file.extractall(path=working_directory)
+        if not os.path.isdir(filename):
+            # if it's not a directory, we consider it's a zip, and we unzip it
+            working_directory = os.path.dirname(filename)
+            zip_file = zipfile.ZipFile(filename)
+            zip_file.extractall(path=working_directory)
+        else:
```

Comment:
not really up to date on tyr, but what if we get a dir that contains only a zip file?